### ISSUE TITLE
Security Vulnerability

### ISSUE THREAD

**Issue by ecmulli** [rstudio/crosstalk#53]
My company uses Blackduck ( https://www.blackducksoftware.com/ ) to monitor security issues in our R packages and flagged crosstalk with a security vulernability in jQuery Core. The descriptions says: JQuery Core contains a flaw that allows a cross-site scripting attack. This flaw exists because the ajax/scripts.js script does not validate input when handling third-party text before returning it to users. This may allow a remote attacked to create a specially crafted request that would execute arbitrary script code in user's browser session within the trust relationship between their browser and the server. 

Can you address this security issue?

**Comment by jcheng5** [comment]
Can you provide more detail? Does black duck have a link we can reference for this rule?

**Comment by ecmulli** [comment]
I’ve emailed the IT manager who owns the Black Duck process. I’ll get all of the info I can and get it to you as soon as possible. 

Evan Mullins

**Comment by ecmulli** [comment]
Joe,

Here is the all of the detail we have. 



Evan Mullins

**Comment by bschneidr** [comment]
@ecmulli Your message appears to be blank. Could you visit the Github website and edit your comment with the intended details?

**Comment by jcheng5** [comment]
@ecmulli Bump? If you must email attachments, please send directly to security@rstudio.com and include a link to this issue. Thanks.

**Comment by ken-duck** [comment]
I don't know if these are the issues indicated by the report discussed by the original issue author, but there are a number of possible XSS vulnerabilities against the version of [jquery](https://github.com/rstudio/crosstalk/blob/master/inst/lib/jquery/jquery.js) you appear to be using. I can add a list of ones I know about if that helps (or email them if you would rather).

**Comment by maelle** [comment]
For info I found the link to this issue after running [`oysteR::audit_deps_with_oss_index()`](https://github.com/sonatype-nexus-community/oysteR)

```r
[93/534] - pkg:cran/crosstalk@1.0.0 - Total vulnerabilities found: 1
Vulnerabilities detected for this component:

	CWE: CWE-79
	Title: CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
	Description: The software does not neutralize or incorrectly neutralizes user-controllable input before it is placed in output that is used as a web page that is served to other users.
	CVSS Score: 7.2
	CVSS Vector: CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:C/C:L/I:L/A:N
	ID: 49da4413-af2b-4e55-acc0-9c752e30dde4
	Reference: https://ossindex.sonatype.org/vuln/49da4413-af2b-4e55-acc0-9c752e30dde4
```

**Comment by cpsievert** [comment]
I believe this is referencing the security vulnerability introduced vy jQuery 3.5.0, which was fixed in 3.5.1, and crosstalk is now on jQuery 3.5.1.

(please reopen if there's anything else that I'm missing here)
